Composer/packages/client/src/pages/design/createDialogModal.tsx

```diff
@@ -36,6 +36,20 @@ export const CreateDialogModal: React.FC<CreateDialogModalProps> = props => {
   const [formData, setFormData] = useState(initialFormData);
   const [formDataErrors, setFormDataErrors] = useState<{ name?: string }>({});
 
+  useEffect(() => {
+    const handleUserKeyPress = event => {
+      const { keyCode } = event;
+      if (keyCode === 9) {
+        const errors = validateForm(formData);
+        setFormDataErrors(errors);
+      }
+    };
+    document.addEventListener('keydown', handleUserKeyPress);
+    return () => {
+      document.removeEventListener('keydown', handleUserKeyPress);
+    };
+  }, []);
```

Comment:
These should be scope to the inputs or the form element, not the document.

Reply:
Actually, we should just hook into the onChange. Wrap each change handler in a function that validates the form data after it is updated.

Reply:
Can this be removed now since the validation is happening elsewhere?